### PR TITLE
[develop] schedule clean up and error tests

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Removing some duplicate code when handling "when" schedule items. 
Moving tests looking for errors in test_errors.  
Adding new tests to account for other error situations.

### Tests written?
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/schedule.py [hunk 1]
```diff
@@ -911,38 +911,27 @@ def _handle_when(job, data, loop_interval):
...
-                            when__ = dateutil_parser.parse(i)
-                        except ValueError:
-                            data['_error'] = ('Invalid date string {0}. '
-                                              'Ignoring job {1}.'.format(i, job))
-                            log.error(data['_error'])
-                            return data
+                        __when = i
+
+                    try:
+                        when__ = dateutil_parser.parse(__when)
```
Comment (s0undt3ch) [inline salt/utils/schedule.py]: First check if its already a date time object, if not parse.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by garethgreenaway** [comment]
@s0undt3ch :+1: Grabbed the other `isinstance` checks from #46540

**Review by s0undt3ch — APPROVED** [review]
Looks right now :+1:

**Comment by s0undt3ch** [comment]
We did find another issue though. Happening on 2018.3.
If you what 2 when entries, the second never fires because _next_start_time is set to the first fire time and in the checks, since that date is way in the past, compared to now - interval, it never fires the 2nd entry.

I'll submit a PR against 2018.3

**Comment by s0undt3ch** [comment]
Never mind my last comment. I was not updating all of the `_` prefixed fields on our end causing the scheduler to misbehave.

**Pull request merged**